packages/core/src/tracing/trace.ts

```diff
@@ -253,13 +250,14 @@ function createChildSpanOrTransaction({
       traceId,
       parentSpanId,
       parentSampled: sampled,
+      // TODO: We should probably only set this if we know we are continuing a trace
+      // For now, there isn't really a way to know this from in here, though...
       ...spanContext,
-      metadata: {
-        dynamicSamplingContext: dsc,
-        // eslint-disable-next-line deprecation/deprecation
-        ...spanContext.metadata,
-      },
     });
+
+    if (dsc) {
+      freezeDscOnSpan(span, dsc);
+    }
```

Comment:
Just to check here but I think this is already how it works: Freezing only happens where we're continuing a trace (i.e. if a propagationContext _with_ a DSC exists). In case it doesn't (i.e. we're head of trace || no DSC on propCtx), we don't freeze the DSC at all at the moment, correct? Meaning, we'd populate it for outgoing requests headers or when the span/txn is is processed.

Reply:
yes, `dsc` will only be set on the propagation context if we had an incoming trace, otherwise this will be `undefined` and we don't freeze it!